docs/source/miscellaneous/doc/adr/0011-retry-long-running-locks-via-blockingtimeoutexception.md

```diff
@@ -0,0 +1,161 @@
+
+This solution appears to be the cleanest of the "free resources"-based solutions, including the one we chose to
+implement. Unfortunately, while this feature has been requested in Jetty, as at time of writing this has not
+been implemented yet; see [Jetty issue #824](https://github.com/eclipse/jetty.project/issues/824).
+
+#### 5. Have clients truncate individual requests to the idle timeout
+
+An alternative to having the server return `BlockingTimeoutException`s on long-running requests would be for clients
+to trim down any requests to an appropriate length (or, in the case of `BLOCK_INDEFINITELY`, indefinitely send
+requests of a suitable length). For example, with the default idle timeout of 30 seconds, a client wishing to block
+for 45 seconds could send a lock request that blocks for 30 seconds, and upon failure submit another request that
+blocks for just under 15 seconds (suitably accounting for network overheads).
+
+This solution is relatively similar to what was implemented, though it requires clients to know what the
+aforementioned "appropriate length" should be (it needs to be the idle timeout or less) which is inappropriate as
+that timeout is configured on the server side.
+
+#### 6. Implement a "magic" HTTP status code or header to ask clients to retry
+
+An alternative to serializing exceptions into `SerializableError`s could be defining a specific HTTP status code
+and/or custom header to indicate that a blocking timeout has occurred and/or clients should retry. This is used
+in practice e.g. in nginx, where a 495 indicates an error with a client's SSL certificates.
+
+This solution would be simpler than serializing exceptions; our existing `AtlasDbErrorDecoder` already switched on the
+status code returned in an HTTP response. However, we prefer not to introduce any custom status codes where feasible
+(since clients are unlikely to understand these status codes). A similar argument, though perhaps weaker, applies
+for headers as well.
+
+## Consequences
+
+#### BLOCK_FOR_AT_MOST Behaviour
+
+After implementation of the above, lock requests that block for less than the idle timeout, or that are 
+`BLOCK_INDEFINITELY` will behave correctly. However, lock requests that block for more than the idle timeout will
+be incorrect:
+
+1. Client A acquires lock L and repeatedly refreshes it
+2. (T = 0) Client B blocks on L; it wants to block for at most 45 seconds.
+3. (T = 30) Client B receives a `BlockingTimeoutException`, and retries - it blocks again on L for at most 45 seconds
+4. (T = 30k, for positive integer k) Client B receives a `BlockingTimeoutException` and retries on L...
+
+They will actually behave like `BLOCK_INDEFINITELY` requests unless the client retries correctly, by suitably
+reducing the blocking duration of lock requests when retrying. We can implement this by having lock service clients 
+modify their lock requests on retrying, though that should be the subject of a separate implementation and separate
+ADR.
```

Comment:
I'll leave a note here explaining why something that looks as simple as this needs an ADR!